packages/core/src/codewhisperer/service/transformByQ/transformationResultsViewProvider.ts

```diff
@@ -195,36 +215,39 @@ export class DiffModel {
             }
             return []
         })
-
-        return this.changes
+        this.patchFileNodes.push(patchFileNode)
+        return patchFileNode
     }
 
     public getChanges() {
-        return this.changes
+        return this.patchFileNodes.flatMap((patchFileNode) => patchFileNode.children)
     }
 
     public getRoot() {
-        return this.changes[0]
+        return this.patchFileNodes.length > 0 ? this.patchFileNodes[0] : undefined
     }
 
     public saveChanges() {
-        this.changes.forEach((file) => {
-            file.saveChange()
+        this.patchFileNodes.forEach((patchFileNode) => {
+            patchFileNode.children.forEach((changeNode) => {
+                changeNode.saveChange()
+            })
         })
-
-        this.clearChanges()
     }
 
     public rejectChanges() {
         this.clearChanges()
     }
 
     public clearChanges() {
-        this.changes = []
+        this.patchFileNodes = []
+        // transformByQState.setSummaryFilePath('')
+        // transformByQState.setProjectCopyFilePath('')
+        // transformByQState.setResultArchiveFilePath('')
```

Comment:
These 3 commented out lines should already be done in a clean-up method we have in `startTransformByQ.ts`

Reply:
It looks like they're already done in the `reset()` function actually, which also calls `clearChanges`